### PR TITLE
Clicking on the >> WALL << of a fire alarm/light switch with your hand will activate it

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request

Clicking on the turf of a fire alarm/light switch with your hand will activate it, similar to how you can click the turf a door is on to close it and not pixel hunt. Refers to the turf it *looks* like it's on, not the one it actually is on, since fire alarms and light switches are actually on the turf near the wall, pixel shifted up.

False walls are ignored.

## Why It's Good For The Game

Lets our artists do whatever they want to these sprites, or those of future consumers, without worrying about niche balance stuff. I really like people using fire alarms for environmental combat and don't want that to get in the way of spriting.

## Changelog
:cl:
qol: Clicking on the turf of a fire alarm/light switch with your hand will activate it.
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
@Fikou Explain

**Comment by Pickle-Coding** [comment]
Would this redirect clicks from people trying to pick up an item from the turf?

**Comment by Mothblocks** [comment]
No

**Comment by Mothblocks** [comment]
Plus how are you picking things up from a wall anyway

File: code/datums/components/redirect_attack_hand_from_turf.dm [hunk 1]
```diff
@@ -0,0 +1,116 @@
+/// Will redirect touching the turf it is on with your hand to the attack_hand of the parent object.
+/datum/component/redirect_attack_hand_from_turf
+	VAR_PRIVATE
```
Comment (LemonInTheDark) [inline code/datums/components/redirect_attack_hand_from_turf.dm:3]: I hate you

Comment (Mothblocks) [reply code/datums/components/redirect_attack_hand_from_turf.dm:3]: It's been merged before because it's the best way to do this

**Comment by Pickle-Coding** [comment]
> Plus how are you picking things up from a wall with an APC/light switch anyway

Wall deconstruction.

**Comment by zxaber** [comment]
>False walls are ignored.

False walls should break anything hanging on them when they open anyway.

Also, does this carry over to silicon interactions?

**Comment by Mothblocks** [comment]
I don't think so? But I could if that wouldn't cause problems

**Comment by zxaber** [comment]
If the purpose is to solve pixelhunting, it probably should be consistent with silicons too.

Also, what happens if a door is placed where the fire alarm's visible placement was? Does clicking the turf affect both?

**Comment by Mothblocks** [comment]
You would be clicking on the door and not the turf. If you open the door and then click the tile then the order is unspecified.

**Comment by Iamgoofball** [comment]
@Mothblocks you might want to just change the title to say wall instead of turf, people are going to be confused otherwise

**Comment by Mothblocks** [comment]
Ummm good question. I think yes and I will fix that

**Comment by Mothblocks** [comment]
Also Fikou made a good point that airlocks should use the component too

**Comment by Mothblocks** [comment]
Silicon attack doesn't have a signal? O_o

**Comment by Mothblocks** [comment]
In order to make this work on airlocks I have to remove their `INTERACT_MACHINE_REQUIRES_SILICON` interaction flag, which shouldn't exist anyway. Let's see how much this breaks

File: code/game/turfs/open/floor.dm [hunk 2]
```diff
@@ -115,13 +115,6 @@
 /turf/open/floor/attack_paw(mob/user, list/modifiers)
 	return attack_hand(user, modifiers)
 
-/turf/open/floor/attack_hand(mob/user, list/modifiers)
-	. = ..()
```
Comment (Mothblocks) [inline code/game/turfs/open/floor.dm:119]: This already calls signal, this attack_hand was never needed

**Comment by timothymtorres** [comment]
I'm getting runtime spammed from this as soon as I latejoin from the shuttle.

```
[13:29:44] Runtime in stack_trace.dm, line 4: atom_attack_robot overridden. Use override = TRUE to suppress this warning. Target: the plating (/turf/open/floor/plating) Proc: /datum/component/redirect_attack_hand_from_turf/proc/on_attack_hand (code/datums/components/_component.dm:230)
proc name: stack trace (/proc/_stack_trace)
src: null
call stack:
stack trace("atom_attack_robot overridden. ...", "code/datums/components/_compon...", 230)
/datum/component/redirect_atta... (/datum/component/redirect_attack_hand_from_turf): RegisterSignal(the plating (49,135,2) (/turf/open/floor/plating), "atom_attack_robot", /datum/component/redirect_atta... (/datum/component/redirect_attack_hand_from_turf/proc/on_attack_hand), 0)
/datum/component/redirect_atta... (/datum/component/redirect_attack_hand_from_turf): RegisterSignals(the plating (49,135,2) (/turf/open/floor/plating), /list (/list), /datum/component/redirect_atta... (/datum/component/redirect_attack_hand_from_turf/proc/on_attack_hand), 0)
/datum/component/redirect_atta... (/datum/component/redirect_attack_hand_from_turf): connect to new turf()
/datum/component/redirect_atta... (/datum/component/redirect_attack_hand_from_turf): on moved(Arrivals Shuttle Airlock (/obj/machinery/door/airlock/titanium), the plating (174,33,12) (/turf/open/floor/plating), 9, 1, null, 0)
Arrivals Shuttle Airlock (/obj/machinery/door/airlock/titanium): SendSignal("movable_moved", /list (/list))
Arrivals Shuttle Airlock (/obj/machinery/door/airlock/titanium): Moved(the plating (174,33,12) (/turf/open/floor/plating), 9, 1, null, 0)
Arrivals Shuttle Airlock (/obj/machinery/door/airlock/titanium): abstract move(the plating (49,135,2) (/turf/open/floor/plating))
Arrivals Shuttle Airlock (/obj/machinery/door/airlock/titanium): onShuttleMove(the plating (49,135,2) (/turf/open/floor/plating), the plating (174,33,12) (/turf/open/floor/plating), /list (/list), 8, Transit for arrival/arrivals s... (/obj/docking_port/stationary/transit), the arrivals shuttle (/obj/docking_port/mobile/arrivals))
the arrivals shuttle (/obj/docking_port/mobile/arrivals): takeoff(/list (/list), /list (/list), /list (/list), 0, 8, Transit for arrival/arrivals s... (/obj/docking_port/stationary/transit), Hyperspace (/area/shuttle/transit))
the arrivals shuttle (/obj/docking_port/mobile/arrivals): initiate docking(the arrivals (/obj/docking_port/stationary), 8, 0)
the arrivals shuttle (/obj/docking_port/mobile/arrivals): initiate docking(the arrivals (/obj/docking_port/stationary), 8)
the arrivals shuttle (/obj/docking_port/mobile/arrivals): check()
the arrivals shuttle (/obj/docking_port/mobile/arrivals): check()
Shuttle (/datum/controller/subsystem/shuttle): fire(0)
Shuttle (/datum/controller/subsystem/shuttle): ignite(0)
Master (/datum/controller/master): RunQueue()
Master (/datum/controller/master): Loop(2)
Master (/datum/controller/master): StartProcessing(0)
```

**Comment by Mothblocks** [comment]
Oh, they needed to be added to `disconnect_from_old_turf`. I wish we had a more automatic solution to this.

**Pull request merged**